dot/network/service.go

```diff
@@ -691,16 +690,15 @@ func (s *Service) ReportPeer(change peerset.ReputationChange, p peer.ID) {
 }
 
 func (s *Service) startPeerSetHandler() {
-	s.host.cm.peerSetHandler.Start(s.ctx)
+	s.host.cm.peerSetHandler.Start(s.ctx, s)
```

Comment:
That's mega strange :smile: We pass the service context AND the service to a sub-sub-sub field of the service... 

Fine with the context, but passing the service itself is just weird i think